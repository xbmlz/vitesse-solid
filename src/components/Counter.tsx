import { Button, VStack } from '@hope-ui/core'
import { createSignal } from 'solid-js'

export default function Counter() {
  const [count, setCount] = createSignal(0)
  return (
    <VStack spacingY={2}>
      <div class="i-carbon-dicom-overlay text-4xl" />
      <div class="text-2xl">About</div>
      <Button variant="solid" size="sm" class="my-4" onClick={() => setCount(count() + 1)}>
        Clicks: {count()}
      </Button>
    </VStack>
  )
}
