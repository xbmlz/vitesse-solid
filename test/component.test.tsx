import { describe, expect, it } from 'vitest'
import { fireEvent, render } from '@solidjs/testing-library'
import Counter from '../src/components/Counter'

describe('<Counter />', () => {
  it('increments value', async () => {
    const { queryByRole, unmount } = render(() => <Counter />)
    const button = (await queryByRole('button')) as HTMLButtonElement
    expect(button).toBeInTheDocument()
    expect(button).toHaveTextContent(/Clicks: 0/)
    fireEvent.click(button)
    expect(button).toHaveTextContent(/Clicks: 1/)
    unmount()
  })

  it('renders 1', () => {
    const { container, unmount } = render(() => <Counter />)
    expect(container).toMatchSnapshot()
    unmount()
  })
})
