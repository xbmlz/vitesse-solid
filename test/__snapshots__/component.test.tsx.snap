// Vitest Snapshot v1

exports[`<Counter /> > renders 1 1`] = `
<div>
  <div
    class="hope-c-PJLV hope-c-PJLV-icLarnM-css hope-Stack-root"
  >
    <div
      class="i-carbon-dicom-overlay text-4xl"
    />
    <div
      class="text-2xl"
    >
      About
    </div>
    <button
      class="hope-Button-root hope-c-gcZMyU hope-c-ezsQXO hope-c-hVhtWI my-4"
      type="button"
    >
      Clicks: 
      0
    </button>
  </div>
</div>
`;
